### ISSUE TITLE
HGFS error in Vagrant after building image on Atlas via packer

### ISSUE THREAD

**Issue by clubdeuce** [hashicorp/packer#3502]
After doing a `packer push` with a `vmware-iso` builder and installing the vmware-tools during the build process, I **always** get the HGFS module not loaded error in Vagrant after downloading the new box image from Atlas.

Is this due to different versions of VMWare on my host machine and on the packer build servers on Atlas? I am using packer 0.10.0 and VMWare Fusion Version 8.1.1 (3771013).

**Comment by cbednarski** [comment]
> Is this due to different versions of VMWare on my host machine and on the packer build servers on Atlas?

Probably not. The HGFS module has to be loaded inside the VM before VMware on your machine can use it, so if the HGFS module is not loading it is an issue with either the VMware tools installer or the kernel inside the VM. As far as I know HGFS support has not changed significantly since Fusion 8 / Workstation 12 was released so I don't think Fusion 8.1.1 makes any difference here.

For some reason, the VMware tools installer doesn't work with newer OSes / kernels. There is an open source package called open-vm-tools supported by VMware which works with newer kernels but **it does not include the HGFS module**. To get HGFS working I had to both install open-vm-tools _and_ run the vmware tools installer. Here's an [example for Ubuntu 16.04](https://github.com/cbednarski/packer-ubuntu/blob/master/scripts-1604/vmware.sh).

So basically:

1. Install `open-vm-tools` to get the latest open source version that is compiled for your kernel / OS (this is maintained by VMware)
2. Install compiler, kernel headers, and VMware tools using the vmware tools installer to add the HGFS kernel module (this is not included in open-vm-tools).
3. Make sure the HGFS kernel module is loaded when the OS boots.

Note also that HGFS is only required for shared folders in vagrant. If you don't plan to use that feature you don't need to go through all this and you can just use the `open-vm-tools` package instead, which is much easier to install if it's available.

**Comment by clubdeuce** [comment]
@Cbednarski Thank you for your reply. 

However, everything works as expected if I do `packer build` on my local machine and then upload the images to Atlas. The issue is specific to builds on the Atlas servers.

**Comment by cbednarski** [comment]
@clubdeuce Can you email us at support@hashicorp.com so we can get some more info and double-check the version we're running? Feel free to include a link to this ticket. Thanks!

**Comment by clubdeuce** [comment]
@cbednarski I can.

**Comment by SwampDragons** [comment]
Closing since Atlas is deprecated.
